src/services/quandl-service.js

```diff
@@ -113,16 +163,42 @@
             });
         }
 
-        getData(stockCode, cb) {
+        /**
+        * @param stockCode {String} Stock code to query
+        * @param cb {Function} callback to be called on success and error
+        * @todo use alternative API key instead of defaulting to No key
+        * @todo should we show a warning to the user when we swap to anonymous?
+        */
+        getData(stockCode, cb, isRetry = false) {
             return this.stockData().get({ code: stockCode }, (result) => {
-                var stock = {
-                    name: result.dataset.name,
+                if (!useApiKey && !isRetry) {
+                    this._notify('CONNECTION_STATUS_CHAGED', (useApiKey = true));
+                }
+                cb({
+                    success: true,
                     code: stockCode,
+                    name: result.dataset.name,
                     data: result.stockData.data
-                };
-
-                cb(stock);
-            });
+                });
+            }, (request) => {
+                // only use the failsafe once per call
+                if (!isRetry) {
+                    this._notify('CONNECTION_STATUS_CHAGED', (useApiKey = false));
+                    this.getData(stockCode, cb, true);
+                } else {
+                    // pass data on so an error message can be shown
+                    cb({
+                        success: false,
+                        code: request.status,
```

Comment:
Overall I don't see an issue; One is a stock the other is an error.
I understand how "details" might be too generic as a field. Especially since it can be hard to know which code is which (http vs quandl), same for message.
Will rename fields to be more specific and replace "details" with "data"